src/components/FullscreenModal/Header/useStickyHeader.ts

```diff
@@ -0,0 +1,43 @@
+import { useCallback, useEffect, useState } from 'react';
+import debounce from 'lodash.debounce';
+
+export const useStickyHeader = (
+  modalRef: React.MutableRefObject<HTMLElement>,
+  modalHeaderRef: React.MutableRefObject<HTMLElement>,
+): { isFixed: boolean } => {
+  const [isFixed, setIsFixed] = useState(false);
+
+  const isInView = useCallback(() => {
+    if (modalRef.current === null || modalHeaderRef.current === null) return;
+    const scrollOffset = modalRef.current.scrollTop;
+    const headerHeight = modalHeaderRef.current.scrollHeight;
+
+    if (isFixed && scrollOffset <= headerHeight) {
+      setIsFixed(false);
+    } else if (!isFixed && scrollOffset > headerHeight) {
+      setIsFixed(true);
+    }
+  }, [isFixed, setIsFixed, modalRef, modalHeaderRef]);
+  const [observer] = useState(new MutationObserver(() => isInView()));
```

Comment:
For this block the same thing as `useStickyHeader` works, e.g. use `useRef` to keep mutation observer instance, disconnect old mutation observers whenever the useEffect body is executed, and clean old listener on unmount and useEffect executions